packages/react-dom/src/events/DOMModernPluginEventSystem.js

```diff
@@ -165,6 +167,57 @@ export const reactScopeListenerStore: WeakMap<
   >,
 > = new PossiblyWeakMap();
 
+function executeDispatch(
+  event: ReactSyntheticEvent,
+  listener: Function,
+  currentTarget: EventTarget,
+): void {
+  const type = event.type || 'unknown-event';
+  event.currentTarget = currentTarget;
+  invokeGuardedCallbackAndCatchFirstError(type, listener, undefined, event);
+  event.currentTarget = null;
+}
+
+function executeDispatchesInOrder(event: ReactSyntheticEvent): void {
+  // TODO we should remove _dispatchListeners and _dispatchInstances at some point.
+  const dispatchListeners = event._dispatchListeners;
+  const dispatchInstances = event._dispatchInstances;
+  let previousInstance;
+
+  if (dispatchListeners !== null && dispatchInstances !== null) {
+    for (let i = 0; i < dispatchListeners.length; i++) {
+      const dispatchInstance = dispatchInstances[i];
+      const dispatchListener = dispatchListeners[i];
+      let instance = dispatchInstance;
+      let currentTarget;
+
+      // dispatchInstance can come in two forms:
```

Comment:
I find this confusing. I worry these special cases are making modern event system harder to follow than the legacy one. Imagine we add a `container` field to Fiber later and this will brek.

If we really need to keep more data here, can we always wrap them in objects?

Reply:
Yeah, I see that. I'll revise.

Reply:
Or I guess a new field so that we don't put more things into `_dispatchInstances`. Like `_dispatchContainers`.

Reply:
I was doing just that :)